src/flags.ts

```diff
@@ -43,11 +44,25 @@ const defaultValues = {
       flags[FlagNames.GIT_CHANGE_TARGET] = process.env.CHANGE_TARGET
     }
   },
+  [FlagNames.BUILD_SCRIPT]: async (flags: any) => {
+    if (flags[FlagNames.ARCHETYPE] && !flags[FlagNames.BUILD_SCRIPT]) {
+      flags[FlagNames.BUILD_SCRIPT] = resolve(__dirname, `./archetypes/${flags[FlagNames.ARCHETYPE]}/build`)
+    } else {
+      flags[FlagNames.BUILD_SCRIPT] = `${relative(process.cwd(), '.pipeline/lib/build')}`
+    }
+  },
+  [FlagNames.DEPLOY_SCRIPT]: async (flags: any) => {
+    if (flags[FlagNames.ARCHETYPE] && !flags[FlagNames.DEPLOY_SCRIPT]) {
+      flags[FlagNames.DEPLOY_SCRIPT] = resolve(__dirname, `./archetypes/${flags[FlagNames.ARCHETYPE]}/deploy`)
+    } else {
+      flags[FlagNames.DEPLOY_SCRIPT] = `${relative(process.cwd(), '.pipeline/lib/deploy')}`
+    }
+  },
 }
 
 export const flagConfigScript = flags.string({name: FlagNames.CONFIG_SCRIPT, hidden: true, default: `${relative(process.cwd(), '.pipeline/lib/config.js')}`})
-export const flagBuildScript = flags.string({name: FlagNames.BUILD_SCRIPT, hidden: true, default: `${relative(process.cwd(), '.pipeline/lib/build.js')}`})
-export const flagDeployScript = flags.string({name: FlagNames.DEPLOY_SCRIPT, hidden: true, default: `${relative(process.cwd(), '.pipeline/lib/deploy.js')}`})
+export const flagBuildScript = flags.string({name: FlagNames.BUILD_SCRIPT, hidden: true})
+export const flagDeployScript = flags.string({name: FlagNames.DEPLOY_SCRIPT, hidden: true})
```

Comment:
Are any applications that wish to use nrdk for build and deploy not needed to provide their build and deploy js?

Reply:
Yes, but we are starting to identify pattern and bundle them as archetype, so that they don't need to provide the build/deployment scripts (or OpenShift templates) anymore.